src/Stratis.Bitcoin/Utilities/Extensions/CollectionExtensions.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Stratis.Bitcoin.Utilities.Extensions
+{
+    /// <summary>
+    /// Extension methods for collections.
+    /// </summary>
+    [DebuggerStepThrough()]
+    public static class CollectionExtensions
```

Comment:
Why was this class added? It's not used anywhere and most likely outside of the scope of this pr.

Reply:
agree

Reply:
was used and then usage was removed but still useful to have.